packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -979,10 +979,45 @@ async function handler(
 
   const allCredentials = await getAllCredentials(organizerUser, eventType);
 
-  const isOrganizerRescheduling = organizerUser.id === userId;
+  let requiresConfirmation = eventType?.requiresConfirmation;
+  const rcThreshold = eventType?.metadata?.requiresConfirmationThreshold;
+  if (rcThreshold) {
+    if (dayjs(dayjs(reqBody.start).utc().format()).diff(dayjs(), rcThreshold.unit) > rcThreshold.time) {
+      requiresConfirmation = false;
+    }
+  }
+
+  /**
+   * Tells that the user doing rescheduling is logged in as the earlier booking owner
+   *
+   * If the user is not the owner of the event, new booking can possibly be pending
+   * Otherwise, an owner rescheduling should be always accepted.
+   * Before comparing make sure that userId is set, otherwise undefined === undefined
+   */
+  const userReschedulingIsOwner = userId && originalRescheduledBooking?.user?.id === userId;
+
+  /**
+   * Tells that the user doing booking(fresh or reschedule) is logged in as the organizer of the event
+   */
+  const isBeingBookedByOrganizer = organizerUser.id === userId;
+
+  const canBookWithoutConfirmation = isBeingBookedByOrganizer || userReschedulingIsOwner;
+  const isConfirmedByDefault = (!requiresConfirmation && !paymentAppData.price) || canBookWithoutConfirmation;
+
+  if (canBookWithoutConfirmation) {
+    loggerWithEventDetails.debug(
+      "user can reschedule without confirmation",
+      safeStringify({
+        isOrganizerRescheduling: isBeingBookedByOrganizer,
+        userReschedulingIsOwner,
+        originalRescheduledBooking,
+        isConfirmedByDefault,
+      })
+    );
+  }
```

Comment:
Placed all requiresConfirmation related logic here as it was becoming really difficult to debug when these variables were being set.